WordPress/gutenberg
# Pattern Explorer Modal: Select the first category as the initial category

Fixes #47657



## What?

If no inserter sidebar pattern category has been selected, make sure the first pattern category is selected as the initially selected category in the pattern explorer modal.

## Why?

This seems to be a regression, where there's no pattern category selected when the modal first opens so the whole thing is empty. This is only if no pattern category has been selected in the inserter sidebar.

## How?

Take the first category object in the `categories` array and pass that as `initialCategory`. 

## Testing Instructions




1. Open the post or site editor.
2. Open the inserter sidebar
3. Select "Explore all patterns" button at the bottom of the sidebar
4. Confirm the first pattern category is selected when the modal opens.
5. Close the modal and select a pattern category in the inserter sidebar.
6. Select the "Explore all patterns" button again, and confirm that the same pattern category is selected in the modal.

### Testing Instructions for Keyboard


## Screenshots or screencast 
**Before**:
<img width="1201" alt="Screenshot 2023-02-01 at 4 06 21 PM" src="https://user-images.githubusercontent.com/1464705/216197016-287e3799-59f1-414f-8b3a-591da1e821f8.png">

**After:**
<img width="1225" alt="Screenshot 2023-02-01 at 4 05 47 PM" src="https://user-images.githubusercontent.com/1464705/216196975-648f800e-6162-4f37-90e1-8b42995e6671.png">
